src/signers.rs

```diff
@@ -1,16 +1,103 @@
+use std::{path::PathBuf, str::FromStr};
+
 use bdk::{
     bitcoin::{
-        secp256k1::{All, Secp256k1},
-        util::psbt::PartiallySignedTransaction,
+        secp256k1::{self, All, Secp256k1},
+        util::{
+            bip32::{self, ChildNumber, DerivationPath, ExtendedPrivKey},
+            psbt::PartiallySignedTransaction,
+        },
         Network,
     },
     wallet::signer::{Signer, SignerError, SignerId},
 };
-use core::str::FromStr;
-use std::path::PathBuf;
+use miniscript::bitcoin::{PrivateKey, PublicKey};
 
 use crate::cmd::{display_psbt, read_yn};
 
+#[derive(Debug)]
+pub struct XKeySigner {
+    /// The derivation path
+    pub path: bip32::DerivationPath,
```

Comment:
I think we discussed this but I think we should remove the `path` here and just sign anything the key can sign. i.e. assume it's a master key.